warnings on mingw build

See here: https://ci.appveyor.com/project/splinter-build/splinter/builds/27084756/job/g6ehy0wbpqlnohed
```
./src/ninja.cc: In member function 'int {anonymous}::NinjaMain::ToolDeps(const {anonymous}::Options*, int, char**)':
./src/ninja.cc:514:12: warning: unknown conversion type character 'l' in format [-Wformat=]
  514 |     printf("%s: #deps %d, deps mtime %" PRId64 " (%s)\n",
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./src/ninja.cc:514:12: warning: format '%s' expects argument of type 'char*', but argument 4 has type 'TimeStamp' {aka 'long long int'} [-Wformat=]
./src/ninja.cc:514:12: warning: too many arguments for format [-Wformat-extra-args]
./src/ninja.cc:514:12: warning: unknown conversion type character 'l' in format [-Wformat=]
./src/ninja.cc:514:12: warning: format '%s' expects argument of type 'char*', but argument 4 has type 'TimeStamp' {aka 'long long int'} [-Wformat=]
./src/ninja.cc:514:12: warning: too many arguments for format [-Wformat-extra-args]
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,7 @@ endif()
 
 #Fixes GetActiveProcessorCount on MinGW
 if(MINGW)
-target_compile_definitions(libninja PRIVATE _WIN32_WINNT=0x0601)
+target_compile_definitions(libninja PRIVATE _WIN32_WINNT=0x0601 __USE_MINGW_ANSI_STDIO=1)
 endif()
 
 # Main executable is library plus main() function.

--- a/configure.py
+++ b/configure.py
@@ -356,7 +356,7 @@ else:
     except:
         pass
     if platform.is_mingw():
-        cflags += ['-D_WIN32_WINNT=0x0601']
+        cflags += ['-D_WIN32_WINNT=0x0601', '-D__USE_MINGW_ANSI_STDIO=1']
     ldflags = ['-L$builddir']
     if platform.uses_usr_local():
         cflags.append('-I/usr/local/include')

